identity-apps-core/apps/recovery-portal/src/main/webapp/recovery.jsp

```diff
@@ -129,57 +175,168 @@
     }
 
     if (isUsernameRecovery) {
-        // Username Recovery Scenario
+        // Username recovery scenario.
+        String recoveryStage = IdentityManagementEndpointUtil.getStringValue(request.getAttribute("recoveryStage"));
```

Comment:
Encode request attributes

Reply:
We don't need to encode the request attributes as the end users can't change those since they are only used in the server.